newIDE/app/src/GameDashboard/LeaderboardAdmin/index.js

```diff
@@ -0,0 +1,820 @@
+      )
+    );
+    if (!answer) return;
+
+    disableActions(true);
+    setApiError(null);
+    try {
+      await deleteLeaderboardEntry(entryId);
+    } catch (err) {
+      console.error(`An error occurred when deleting entry: ${err}`);
+      setApiError({
+        action: 'entryDeletion',
+        message: (
+          <Trans>
+            An error occurred when deleting the entry, please try again.
+          </Trans>
+        ),
+        itemId: entryId,
+      });
+    } finally {
+      disableActions(false);
+    }
+  };
+
+  React.useEffect(
+    () => {
+      if (isEditingName && newNameTextFieldRef.current) {
+        newNameTextFieldRef.current.focus();
+      }
+    },
+    [isEditingName]
+  );
+
+  React.useEffect(
+    () => {
+      if (leaderboards === null) {
+        _listLeaderboards();
+      }
+    },
+    [leaderboards, _listLeaderboards]
+  );
+
+  const onCopy = React.useCallback(
+    () => {
+      if (!currentLeaderboard) return;
+      // TODO: use Clipboard.js, after it's been reworked to use this API and handle text.
+      navigator.clipboard.writeText(currentLeaderboard.id);
+    },
+    [currentLeaderboard]
+  );
+  if (!isOnline) {
+    return (
+      <CenteredError>
+        <PlaceholderError>
+          <Trans>
+            An internet connection is required to administrate your game's
+            leaderboards.
+          </Trans>
+        </PlaceholderError>
+      </CenteredError>
+    );
+  }
+  if (apiError && apiError.action === 'leaderboardCreation') {
+    return (
+      <CenteredError>
+        <AlertMessage kind="error">{apiError.message}</AlertMessage>
+      </CenteredError>
+    );
+  }
+  if (apiError && apiError.action === 'leaderboardsFetching') {
+    return (
+      <CenteredError>
+        <PlaceholderError onRetry={_listLeaderboards} kind="error">
+          {apiError.message}
+        </PlaceholderError>
+      </CenteredError>
+    );
+  }
+  if (leaderboards === null) {
+    if (isRequestPending) return <PlaceholderLoader />;
+    else {
+      return (
+        <CenteredError>
+          <PlaceholderError onRetry={_listLeaderboards} kind="error">
+            <Trans>
+              An error occurred when retrieving leaderboards, please try again
+              later.
+            </Trans>
+          </PlaceholderError>
+        </CenteredError>
+      );
+    }
+  }
+
+  if (!!leaderboards && leaderboards.length === 0)
+    return (
+      <Line noMargin expand justifyContent="center" alignItems="center">
+        <EmptyPlaceholder
+          title={<Trans>Create your game's first leaderboard</Trans>}
+          description={<Trans>Leaderboards help retain your players</Trans>}
+          actionLabel={<Trans>Create a leaderboard</Trans>}
+          onAdd={() => {
+            _createLeaderboard();
+          }}
+          isLoading={isRequestPending}
+        />
+      </Line>
+    );
+
+  const leaderboardDescription = (
```

Comment:
nit: this is a function, so it should begin with a verb (render, build, create...)